elytron/WFLY-16793-oidc-identity-propagation.adoc

```diff
@@ -0,0 +1,166 @@
+work.
+
+* A new `virtual-security-domain` resource will be introduced in the `elytron` subsystem.
+The new resource will make it possible to outflow security identities established by virtual
+security domains to other security domains. The new `virtual-security-domain` resource will have
+two attributes:
+
+** `name` - This is the runtime name of a deployment associated with a virtual security domain (e.g.,
+`DEPLOYMENT_NAME.ear`, a deployment that has a subdeployment that is secured using OIDC).
+
+** `outflow-security-domains` - This is the list of `security-domains` that security identities from
+the virtual security domain should be automatically outflowed to. This will be a required attribute
+and must be non-empty.
+
+** As an example, a `virtual-security-domain` could be added as follows:
+[source]
+----
+/subsystem=elytron/virtual-security-domain=DEPLOYMENT_NAME.ear:add(outflow-security-domains=...)
+----
+
+* A new `trusted-virtual-security-domains` attribute will be added to the `security-domain` resource
+in the `elytron` subsystem. This is the list of runtime names of deployments associated with
+virtual security domains that should be trusted by a `security-domain`. This will be an optional
+attribute.
+** e.g., `trusted-virtual-security-domains=DEPLOYMENT_NAME.ear` indicates that this `security-domain`
+should trust the virtual security domain associated with the `DEPLOYMENT_NAME.ear` deployment.
+
+Note that in general, we are following the same concept of security outflow and trust that
+is used when propagating identities between traditional Elytron `security-domains`. Here, we
+are simply extending this concept for the case where we need to propagate an identity from
+a virtual security domain to a traditional `security-domain`.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+Note that in the future, we might want to consider adding other attributes to the `virtual-security-domain`
+resource as well. We won't consider this as part of this RFE though.
+
+== Backwards Compatibility
+
+No backwards compatibility concerns.
+
+=== Default Configuration
+
+No changes to the default configuration.
+
+=== Importing Existing Configuration
+
+N/A
+
+=== Deployments
+
+N/A
+
+=== Interoperability
+
+N/A
+
+//== Implementation Plan
+////
+Delete if not needed. The intent is if you have a complex feature which can 
+not be delivered all in one go to suggest the strategy. If your feature falls 
+into this category, please mention the Release Coordinators on the pull 
+request so they are aware.
+////
+
+== Security Considerations
+
+This is a security RFE.
+
+////
+Identification if any security implications that may need to be considered with this feature
+or a confirmation that there are no security implications to consider.
+////
+
+== Test Plan
+
+Elytron subsystem parsing and transformer tests will be added.
+
+Tests will be added to the WildFly testsuite to verify that a security identity
```

Comment:
Are there any other subsystems/resources establishing virtual domains? If it's the case, are those somehow specific and require more tests?

Reply:
The only other place where a virtual security domain is used is for MP JWT. We can add some sanity tests for this case as well. I'll update the analysis.